### PR TITLE
[Agent] postgresql support parse more block and support prepare

### PR DESCRIPTION
**Pull request by qq906907952**
statement

### This PR is for:
- Agent

#### Checklist
- [ ] Added unit test.
#### Backport to branches
- main

### CODE REVIEW

**Review by sharang — APPROVED** [review]

File: agent/src/flow_generator/protocol_logs/sql/postgresql.rs [hunk 1]
```diff
@@ -184,114 +212,174 @@ impl L7ProtocolParserInterface for PostgresqlLog {
...
-    payload: len - 4 byte
-    */
-
     fn parse(&mut self, payload: &[u8]) -> Result<()> {
-        if payload.len() < 5 {
-            return Err(Error::L7ProtocolUnknown);
-        }
+        let mut offset = 0;
+        loop {
+            if offset > payload.len() {
```
Comment (rvql) [inline agent/src/flow_generator/protocol_logs/sql/postgresql.rs:231]: `>=`?

File: agent/src/flow_generator/protocol_logs/sql/postgresql.rs [hunk 2]
```diff
@@ -184,114 +212,174 @@ impl L7ProtocolParserInterface for PostgresqlLog {
...
+    }
 
-        let data_len = read_u32_be(&payload[1..5]);
-        if payload.len() - 1 < data_len as usize {
-            return Err(Error::L7ProtocolUnknown);
+    fn check_is_ssl_req(&self, payload: &[u8]) -> bool {
+        if payload.len() != 8 {
+            return false;
         }
+        self.info.msg_type == LogMessageType::Request && read_u64_be(payload) == SSL_REQ
```
Comment (rvql) [inline agent/src/flow_generator/protocol_logs/sql/postgresql.rs:256]: `payload.len() == 8 && self.info.msg_type == ...`

File: agent/src/flow_generator/protocol_logs/sql/postgresql.rs [hunk 3]
```diff
@@ -184,114 +212,174 @@ impl L7ProtocolParserInterface for PostgresqlLog {
...
+                self.info.context = String::from_utf8_lossy(&data[2..len - 2]).to_string();
+                self.info.ignore = false;
+                Ok(())
+            }
+            'B' | 'F' | 'C' | 'D' | 'H' | 'S' | 'X' | 'd' | 'c' | 'f' => Ok(()),
+            _ => Err(Error::L7ProtocolUnknown),
+        }
+    }
+
+    fn on_resp_block(&mut self, tag: char, data: &[u8]) -> Result<()> {
```
Comment (rvql) [inline agent/src/flow_generator/protocol_logs/sql/postgresql.rs:283]: 这些处理函数可以加一些单元测试

File: agent/src/flow_generator/protocol_logs/sql/postgresql.rs [hunk 4]
```diff
@@ -333,45 +421,79 @@ case 'c':        Copy Done
...
+const RESP_STR_N: &'static str = "No Data";
+const RESP_STR_PARAM_DESC: &'static str = "Parameter Description";
+const RESP_STR_D: &'static str = "Data Row";
+const RESP_STR_G: &'static str = "Start Copy In";
+const RESP_STR_H: &'static str = "Start Copy Out";
+const RESP_STR_W: &'static str = "Start Copy Both";
+const RESP_STR_COPY_DATA: &'static str = "Copy Data";
+const RESP_STR_COPY_DONE: &'static str = "Copy Done";
+
+fn get_response_result(typ: char) -> Option<String> {
```
Comment (rvql) [inline agent/src/flow_generator/protocol_logs/sql/postgresql.rs:443]: 可以返回`Option<&'staic str>`，下同

File: agent/src/flow_generator/protocol_logs/sql/postgresql.rs (hunk 1 above)
Comment (qq906907952) [reply agent/src/flow_generator/protocol_logs/sql/postgresql.rs]: 其实不用等于也可以，不过我还是改了

File: agent/src/flow_generator/protocol_logs/sql/postgresql.rs (hunk 4 above)
Comment (qq906907952) [reply agent/src/flow_generator/protocol_logs/sql/postgresql.rs]: 改成返回 &'staic str

**Review by rvql — APPROVED** [review]

**Pull request merged**